safe_transaction_service/history/views.py

```diff
@@ -500,24 +354,21 @@ def list(self, request, *args, **kwargs):
             200: _schema_200_response,
             422: "code = 1: Checksum address validation failed",
         },
-        manual_parameters=[
-            _schema_executed_param,
-            _schema_queued_param,
-            _schema_trusted_param,
-        ],
     )
     def get(self, request, *args, **kwargs):
         """
-        Returns all the transactions for a given Safe address. The list has different structures depending on the
-        transaction type:
-        - Multisig Transactions for a Safe. `tx_type=MULTISIG_TRANSACTION`. If the query parameter `queued=False` is
-        set only the transactions with `safe nonce < current Safe nonce` will be displayed. By default, only the
-        `trusted` transactions will be displayed (transactions indexed, with at least one confirmation or proposed
-        by a delegate). If you need that behaviour to be disabled set the query parameter `trusted=False`
+        Returns all the *executed* transactions for a given Safe address.
+        The list has different structures depending on the transaction type:
+        - Multisig Transactions for a Safe. `tx_type=MULTISIG_TRANSACTION`.
         - Module Transactions for a Safe. `tx_type=MODULE_TRANSACTION`
         - Incoming Transfers of Ether/ERC20 Tokens/ERC721 Tokens. `tx_type=ETHEREUM_TRANSACTION`
-          Only 1000 newest transfers will be returned.
-        Ordering_fields: ["execution_date"] eg: `execution_date` or `-execution_date`
+        Ordering_fields: ["timestamp"] eg: `-timestamp` (default one) or `timestamp`
+
+        Note: This endpoint has a bug that will be fixed in next versions of the endpoint. Pagination is done
```

Comment:
Yes... But this is something already happening with `ModuleTransactions`, so this is an ongoing bug

Reply:
You are right. Anyway, would it be possible to invalidate the cache in case of detecting a reorg? 

Although maybe the performance gain won't be very big.

Reply:
I think you are mixing threads 😉 . But yes, I wouldn't change more things in this PR. We can always come back and optimize it more in the future

Reply:
Oops, yeah... I have confused the thread. Sorry. It would be for [this](https://github.com/safe-global/safe-transaction-service/pull/2166#discussion_r1703945967).